src/templates/contentPage.js

```diff
@@ -67,10 +113,83 @@ class ContentPage extends Component {
   buildPageHTML(pageContentHTML) {
     return { __html: pageContentHTML };
   }
+  
+  //using the page list, create the grid of cards
+  mapPageListDataToElem() {
+    let row = [], 
+    pageListGrid = [],
+    max = 2,
+    titleCard,
+	themePalette;
+
+    // list needs to be in grid, max 2 per row
+    for (let i = 0; i < max; i+=2) {
+      row = [];
+	  
+	  titleCard = this.props.data.contentfulPage.previousPage;
+	  if (titleCard !== null){
+		themePalette = this.themePaletteToCss(this.props.data.contentfulPage.previousPage.parentMenu.themePalette)
+		row.push(this.pageItemToCard(titleCard, themePalette));
+	  }
+	  
+	  titleCard = this.props.data.contentfulPage.nextPage;
+	  if (titleCard !== null){
+		themePalette = this.themePaletteToCss(this.props.data.contentfulPage.nextPage.parentMenu.themePalette)
+	    row.push(this.pageItemToCard(titleCard, themePalette));
+	  }
+
+      pageListGrid.push(
+        <div className="columns">
+          { row.map(col => { return(<div className="column">{col}</div>) }) }
+        </div>
+      );
+    }
+
+    return pageListGrid;
+  }
+  
+  //generate a card from a page
+  pageItemToCard(page, themePalette) {
+    const url = `/${page.parentMenu.slug}/${page.slug}`;
+    const title = this.getCardTitleForLanguage(page.titleList, this.state.language);
+
+    return (
+      <div className={`content ${themePalette.cardParentContainerHover}`}>
+        <div className={`card ${cardSize} ${themePalette.background}`}>
+          <div className={`card-content ${cardTitleText} ${vertCentred}`}>
+              <Link className={`title ${themePalette.text}`} to={url}>
+                {this.getIconForPage(page)}
+                <br/>
+                {title}
+              </Link>
+          </div>
+        </div>
+      </div>
+    )
+  }
+  
+  //given a page and menu title, load the relevent icon
+  getIconForPage(page) {
+    let imgImp = "#";
+
+    //try to get the image
+    try {
+      imgImp = require(`../content/icons/${page.parentMenu.slug}/${page.slug}.png`);
+    } catch (e){
+      console.log("file " + `/src/content/icons/${page.parentMenu.slug}/${page.slug}.png does not exist` ); 
```

Comment:
logging to the console in this way is not best practice. Try catch is generally for catching exceptions and handling them appropriately, which JavaScript doesn't really do properly haha. I would suggest checking the value for null and assigning a placeholder if it is empty.